### PR TITLE
IDEMPIERE-5355 Improve default trx name (FHCA-3765)

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5355

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/util/Trx.java [hunk 1]
```diff
@@ -131,7 +131,15 @@ public static String createTrxName (String prefix)
...
-		Trx.get(prefix, true);
+		Trx trx = Trx.get(prefix, true);
+		StackTraceElement[] st = new Throwable().fillInStackTrace().getStackTrace();
+		for (StackTraceElement ste : st) {
+			if (! Trx.class.getName().equals(ste.getClassName())) {
+				String displayName = ste.getClassName().concat("_").concat(ste.getMethodName());
+				trx.setDisplayName(displayName);
+				break;
+			}
+		}
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/util/Trx.java:142]: this could be expensive...

Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/compiere/util/Trx.java]: Thanks @hengsin - added the SysConfig TRX_AUTOSET_DISPLAY_NAME (defaults to false).
So, it needs to be set to Y in order to assign the display name based on stack trace.

And also changed the scope, this is just done for transactions without a prefix set.

File: org.adempiere.server/src/main/server/org/compiere/server/TrxInfo.java [hunk 2]
```diff
@@ -57,11 +58,13 @@ public static TrxInfo[] getActiveTransactions() {
 		List<TrxInfo> list = new ArrayList<>();
 		Trx[] trxs = Trx.getActiveTransactions();
 		for (Trx trx : trxs) {
-			if (trx != null && trx.isActive()) {
+			if (trx != null) {
```
[the remark this answers is not in the document]
Comment (CarlosRuiz-globalqss) [reply org.adempiere.server/src/main/server/org/compiere/server/TrxInfo.java]: Ready, added the methods:
getOpenTransactions() and getOpenTransactions(boolean onlyActive)

The original method getActiveTransactions() is still there returning the same array but unused at this moment by core.

File: org.compiere.db.oracle.provider/src/org/compiere/db/DB_Oracle.java [hunk 3]
```diff
@@ -355,7 +355,7 @@ public String getStatus()
             sb.append(" , # Min Pool Size: ").append(m_ds.getMinPoolSize());
             sb.append(" , # Max Pool Size: ").append(m_ds.getMaxPoolSize());
             sb.append(" , # Max Statements Cache Per Session: ").append(m_ds.getMaxStatementsPerConnection());
-            sb.append(" , # Active Transactions: ").append(Trx.getActiveTransactions().length);
+            sb.append(" , # Open Transactions: ").append(Trx.getActiveTransactions().length);
```
Comment (hengsin) [inline org.compiere.db.oracle.provider/src/org/compiere/db/DB_Oracle.java:358]: probably should change to getOpenTransactions() here ?

File: org.compiere.db.postgresql.provider/src/org/compiere/db/DB_PostgreSQL.java [hunk 4]
```diff
@@ -334,7 +334,7 @@ public String getStatus()
             sb.append(" , # Min Pool Size: ").append(m_ds.getMinPoolSize());
             sb.append(" , # Max Pool Size: ").append(m_ds.getMaxPoolSize());
             sb.append(" , # Max Statements Cache Per Session: ").append(m_ds.getMaxStatementsPerConnection());
-            sb.append(" , # Active Transactions: ").append(Trx.getActiveTransactions().length);
+            sb.append(" , # Open Transactions: ").append(Trx.getActiveTransactions().length);
```
Comment (hengsin) [inline org.compiere.db.postgresql.provider/src/org/compiere/db/DB_PostgreSQL.java:337]: probably should change to getOpenTransactions() here ?

File: org.compiere.db.oracle.provider/src/org/compiere/db/DB_Oracle.java (hunk 3 above)
Comment (CarlosRuiz-globalqss) [reply org.compiere.db.oracle.provider/src/org/compiere/db/DB_Oracle.java]: ready - method name was wrong in Trx - fixed and deprecated the method with wrong name

**Review by hengsin — APPROVED** [review]
looks ok

**Pull request merged**